src/App.tsx

```diff
@@ -180,33 +162,41 @@ function App() {
   useEffect(() => {
     async function callToAPIs() {
       try {
-        const [movieResponse, selMovieResponse] = await Promise.all([
+        const [movieResponse] = await Promise.all([
           getMovies(state.search, state.page, mediaType),
           getSelected(selectedID),
         ]);
         update({ type: "movie response set", movieResPayload: movieResponse });
-        update({
-          type: "select movie response set",
-          selMovieResPayload: selMovieResponse,
-        });
-        console.log(selMovieResponse);
       } catch (error) {
         console.log(error);
       }
     }
     if (state.search !== null) {
       callToAPIs();
     }
-  }, [state.search, state.page, selectedID, mediaType]);
+  }, [state.search, state.page, mediaType]);
 
-  if (!state.result) return null;
+  useEffect(() => {
+    async function fetchData() {
+      const selMovieResponse = await getSelected(selectedID);
+
+      update({
+        type: "select movie response set",
+        selMovieResPayload: selMovieResponse,
+      });
+    }
+    fetchData();
+  }, [update, selectedID]);
 
-  update({
-    type: "highest rated movies set",
-    hRMPayload: sortArray(state.result.Search),
-  });
+  const sortedMovies = React.useMemo(() => {
+    return state.result
+      ? state.result.Search.slice(0).sort((a, b) => {
+          return parseInt(a.imdbRating) - parseInt(b.imdbRating);
+        })
+      : [];
```

Comment:
Why is there an empty array rather than null here?

Reply:
i quite like to return the same type if possible so an empty array is still `Movie[]` as opposed to `Movie[] | null`.